src/service/DirWatcher.js

```diff
@@ -0,0 +1,38 @@
+const EventEmitter = require('events');
+const fs = require('fs');
+const path = require('path');
+
+class DirWatcher extends EventEmitter {
+  constructor() {
+    super();
+    this.filesToWatch = [];
+  }
+
+  watch(dataPath, delay) {
+    setInterval(() => this.readDataDir(dataPath), delay * 1000);
+  }
+
+  readDataDir(dataPath) {
+    fs.readdir(dataPath, (err, files) => {
+      if (err) {
+        console.log(err);
+        return;
+      }
+
+      const csvFiles = files.filter((file) => path.extname(file) === '.csv');
+      this.notifyOnDataChanges(csvFiles);
+    });
+  }
+
+  notifyOnDataChanges(currentDataFiles) {
+    currentDataFiles
+      .filter((file) => !this.filesToWatch.includes(file))
```

Comment:
this approach does not allow to detect file content change within already existing file. Moreover, it will not be possible to detect file removal. I'd expect to see at least some notification that a given file has been removed

Reply:
File removal is completely processed here. Otherwise it wouldn't be possible to process file by deleting one and writing file with the same name in the data folder.  This scenario works here.

Changing a file content seems is not in the requirements. I mentioned it in the Readme file. (Note: every CSV file in a directory should be processed only once.) But it's definitely is a point of discussion with a customer. As a solution the md5 sum of the data files could be stored and checked every time to understand that file content is changed. 
For now only adding files, renaming and deleting operations are processed.

Reply:
Indeed, removal is processed. it was only absence of appropriate notification which was confusing. As presence of notification is not clearly stated in the requirement, I guess we could consider requirements as met completely